src/features/pass-survey/model/hooks/useUserActionManager.ts

```diff
@@ -41,12 +50,14 @@ function useUserActionManager({ activityId, activityState }: Args) {
       | SetAnswerAction
       | undefined;
 
-    const shouldUpdateLastAction =
+    const shouldUpdateAllPreviousAnswerActions =
```

Comment:
We need to make sure that we do want to update all the previous `SET_ANSWER` actions in some cases. At the moment, I can't think of an activity item that should correspond to such a rule. Please investigate it and correct me if I am wrong.
If am not wrong then we should always update the last SET_ANSWER action.